tests/proxy/roundtrip_custom_header.t

```diff
@@ -0,0 +1,182 @@
+  $ . ${TESTDIR}/setup_test_env.sh
```

Comment:
The test case for this should be written using `josh-filter` and not using the proxy. Those are easier and faster to run. Proxy tests should only be used for behaviour that is unique to the proxy.

Reply:
Yeah I had wondered about this. I started here because that's how I initially discovered the bug. I do actually have the filter test already written but I was a bit confused by how `--update` and `--reverse` interact and wasn't confident that it was properly reproducing the issue.

I'll swap the tests and then when you re-review perhaps you can verify the approach.

Reply:
I've replaced the test with a filter test but it currently is broken because of my confusion about the above. If you could give some guidance that'd be excellent.

Reply:
Never mind! I took another approach by applying two different but inverse filters instead of using `--reverse`. `:prefix=foo` followed by `:/foo` should actually be closer to what the proxy test was doing anyway.